packages/opentelemetry-plugin-mongodb-core/src/mongodb.ts

```diff
@@ -0,0 +1,246 @@
+
+    if (this._moduleExports.Server) {
+      for (const fn of this._SERVER_METHODS) {
+        this._logger.debug(`patching mongodb-core.Server.prototype.${fn}`);
+        shimmer.wrap(
+          this._moduleExports.Server.prototype,
+          // Forced to ignore due to incomplete typings
+          // tslint:disable-next-line:ban-ts-ignore
+          // @ts-ignore
+          fn,
+          this._getPatchCommand(fn)
+        );
+      }
+    }
+
+    if (this._moduleExports.Cursor) {
+      this._logger.debug(
+        'patching mongodb-core.Cursor.prototype functions:',
+        this._CURSOR_METHODS
+      );
+      shimmer.massWrap(
+        [this._moduleExports.Cursor.prototype],
+        this._CURSOR_METHODS as never[],
+        // tslint:disable-next-line:no-any
+        this._getPatchCursor() as any
+      );
+    }
+
+    return this._moduleExports;
+  }
+
+  /** Unpatches all MongoDB patched functions. */
+  unpatch(): void {
+    shimmer.massUnwrap(
+      [this._moduleExports.Server.prototype],
+      this._SERVER_METHODS as never[]
+    );
+    shimmer.massUnwrap(
+      [this._moduleExports.Cursor.prototype],
+      this._CURSOR_METHODS as never[]
+    );
+  }
+
+  /** Creates spans for Command operations */
+  private _getPatchCommand(operationName: string) {
+    const plugin = this;
+    return (original: Func<mongodb.Server>) => {
+      return function patchedServerCommand(
+        this: mongodb.Server,
+        ns: string,
+        commands: MongoInternalCommand[] | MongoInternalCommand,
+        options: {} | Function,
+        callback: Function
+      ): mongodb.Server {
+        const currentSpan = plugin._tracer.getCurrentSpan();
+        const resultHandler =
+          typeof options === 'function' ? options : callback;
+        if (
+          currentSpan === null ||
+          typeof resultHandler !== 'function' ||
+          typeof commands !== 'object'
+        ) {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const command = commands instanceof Array ? commands[0] : commands;
+        const type = plugin._getCommandType(command, operationName);
+        const span = plugin._tracer.startSpan(`mongodb.${type}`, {
+          parent: currentSpan,
+          kind: SpanKind.CLIENT,
+        });
+        plugin._populateAttributes(
+          span,
+          ns,
+          command,
+          this as MongoInternalTopology
+        );
+        return original.call(
+          this,
+          ns,
+          commands,
+          plugin._patchEnd(span, resultHandler)
+        );
+      };
+    };
+  }
+
+  /**
+   * Get the mongodb command type from the object.
+   * @param command Internal mongodb command object
+   * @param defaulType the default type to return if we could not find a
+   *  specific command.
+   */
+  private _getCommandType(command: MongoInternalCommand, defaulType: string) {
+    if (command.createIndexes !== undefined) {
```

Comment:
should all of these (createIndexes, findAndModify, isMaster, count)  be some enum ?

Reply:
I don't see how you would check against an enum if the property exist ?

Reply:
I'm talking about things you are returning
`return 'createIndexes';`
instead something like 
```
export enum MongoDBCommands {
  CREATE_INDEXES = 'createIndexes',
  FIND_AND_MODIFY = 'findAndModify',
}
```
and then
```
return MongoDBCommands.CREATE_INDEXES
```

Reply:
Oh right i misunderstood that you were speaking about the return type. I can surely use an enum here

Reply:
sry for confusion :)

Reply:
I've ammend by previous commit with your suggestion